app/experimenter/experiments/tests/api/v5/test_serializers.py

```diff
@@ -291,6 +295,54 @@ def test_branch_with_invalid_feature_value_jso(self):
         self.assertFalse(branch_serializer.is_valid())
         self.assertIn("feature_value", branch_serializer.errors)
 
+    def test_branch_update_screenshots(self):
+        branch = NimbusBranchFactory()
+        existing_screenshot = branch.screenshots.first()
+        existing_image = existing_screenshot.image
+        deleted_screenshots = [
+            NimbusBranchScreenshotFactory.create(branch=branch),
+            NimbusBranchScreenshotFactory.create(branch=branch),
+            NimbusBranchScreenshotFactory.create(branch=branch),
+        ]
+
+        updated_screenshot_data = {"id": existing_screenshot.id, "description": "updated"}
+        image_content = TINY_PNG
+        new_screenshot_data = {
+            "description": "new screenshot",
+            "image": SimpleUploadedFile(name="Capture.PNG", content=image_content),
+        }
+        branch_data = {
+            "name": "updated name",
+            "description": "updated description",
+            "ratio": 1,
+            "screenshots": [
+                updated_screenshot_data,
+                new_screenshot_data,
+            ],
+        }
+        branch_serializer = NimbusBranchSerializer(branch, data=branch_data)
+        self.assertTrue(branch_serializer.is_valid(), branch_serializer.errors)
+        branch_serializer.save()
+
+        branch = NimbusBranch.objects.get(pk=branch.id)
+        self.assertEqual(branch.screenshots.count(), 2)
+
+        for screenshot in deleted_screenshots:
+            self.assertFalse(branch.screenshots.filter(pk=screenshot.id).exists())
+
+        screenshots = branch.screenshots.order_by("id")
```

Comment:
I wouldn't depend on this for a stable ordering, I've seen countless teams that the db can reorder inserts/queries depending on the phase of the moon, so see if there's a way you can distinguish some other way?

Reply:
We use it over here in [the `treatment_branches` property](https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L281), so I figured it was convenient enough here.